core/src/main/java/org/axonframework/config/EventProcessingConfiguration.java

```diff
@@ -328,10 +348,21 @@ public EventProcessingConfiguration registerHandlerInterceptor(String processorN
      */
     public List<MessageHandlerInterceptor<? super EventMessage<?>>> interceptorsFor(String processorName) {
         Assert.state(configuration != null, () -> "Configuration is not initialized yet");
-        return handlerInterceptorsBuilders.getOrDefault(processorName, new ArrayList<>())
-                                          .stream()
-                                          .map(hi -> hi.apply(configuration))
-                                          .collect(Collectors.toList());
+
+        List<MessageHandlerInterceptor<? super EventMessage<?>>> interceptors = new ArrayList<>();
+
+        defaultHandlerInterceptors.stream()
+                                  .map(f -> f.apply(configuration, processorName))
+                                  .filter(Objects::nonNull)
+                                  .forEach(interceptors::add);
+
+        handlerInterceptorsBuilders.getOrDefault(processorName, new ArrayList<>())
```

Comment:
Small one, but the static `emptyList` would be a better option here, as it doesn’t create any objects.

Reply:
Good one, will adjust that in `master`, Steven already merged the PR ;)